code/modules/projectiles/projectile/bullets/paintball.dm

```diff
@@ -0,0 +1,44 @@
+/obj/projectile/paintball
+	name = "empty paintball"
+	icon_state = "paintball"
+	damage = 0
+	hitsound = 'sound/weapons/throwtap.ogg'
+
+/obj/projectile/paintball/Initialize(mapload)
+	. = ..()
+	create_reagents(10, NO_REACT)
+
+/obj/projectile/paintball/on_hit(atom/target, blocked = 0, pierce_hit)
+	if(blocked != 100) // not completely blocked
+		..()
+		reagents.trans_to(target, reagents.total_volume, methods = VAPOR)
```

Comment:
You need to use `expose()` like I mentioned in my other suggestion, otherwise you can shoot a beaker with it and it'll fill up

I also recommend you use `VAPOR|TOUCH` and not just `VAPOR`, it's just a bit more sensible when it comes to the actual vector you're using

```suggestion
		reagents.expose(target, methods = VAPOR|TOUCH)
```